superset-frontend/src/components/DynamicPlugins/DynamicPluginProvider.tsx

```diff
@@ -0,0 +1,157 @@
+};
+
+type BeginAction = {
+  type: 'begin';
+  keys: string[];
+};
+
+function pluginContextReducer(
+  state: PluginContextType,
+  action: BeginAction | CompleteAction,
+): PluginContextType {
+  switch (action.type) {
+    case 'begin': {
+      const plugins = { ...state.plugins };
+      action.keys.forEach(key => {
+        plugins[key] = { key, error: null, loading: true };
+      });
+      return {
+        ...state,
+        loading: true,
+        plugins,
+      };
+    }
+    case 'complete': {
+      return {
+        ...state,
+        loading: Object.values(state.plugins).some(
+          plugin => plugin.loading && plugin.key !== action.key,
+        ),
+        plugins: {
+          ...state.plugins,
+          [action.key]: {
+            key: action.key,
+            loading: false,
+            error: action.error,
+          },
+        },
+      };
+    }
+    default:
+      return state;
+  }
+}
+
+export type Props = React.PropsWithChildren<{}>;
+
+const sharedModules = {
+  react: () => import('react'),
+  lodash: () => import('lodash'),
+  'react-dom': () => import('react-dom'),
+  '@superset-ui/chart-controls': () => import('@superset-ui/chart-controls'),
+  '@superset-ui/core': () => import('@superset-ui/core'),
+};
+
+export default function DynamicPluginProvider({ children }: Props) {
+  const [pluginState, dispatch] = useReducer(pluginContextReducer, {
+    // use the dummy plugin context, and override the methods
+    ...dummyPluginContext,
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    fetchAll,
+    loading: isFeatureEnabled(FeatureFlag.DYNAMIC_PLUGINS),
+    // TODO: Write fetchByKeys
+  });
+
+  async function fetchAll() {
+    try {
+      await defineSharedModules(sharedModules);
+      const response = await SupersetClient.get({
+        endpoint: '/dynamic-plugins/api/read',
+      });
+      const plugins: Plugin[] = response.json.result;
+      // const plugins: Plugin[] = [
+      //   {
+      //     name: 'Hello World',
+      //     key: 'superset-chart-hello-world',
+      //     id: 0,
+      //     bundle_url: 'http://127.0.0.1:8080/main.js',
+      //   },
+      // ];
+      dispatch({ type: 'begin', keys: plugins.map(plugin => plugin.key) });
+      await Promise.all(
+        plugins.map(async plugin => {
+          let error: Error | null = null;
+          try {
+            await import(/* webpackIgnore: true */ plugin.bundle_url);
+          } catch (err) {
+            // eslint-disable-next-line no-console
+            console.error(
+              `Failed to load plugin ${plugin.key} with the following error:`,
+              err.stack,
+            );
+            error = err;
+          }
+          dispatch({
+            type: 'complete',
+            key: plugin.key,
+            error,
+          });
+        }),
+      );
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error.stack || error);
```

Comment:
You can `import { logging } from '@superset-ui/core';` and use it as `logging.error(stuff)` and get rid of the `eslint-disable`